clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -402,21 +403,26 @@ else if (strategy == OffsetResetStrategy.LATEST)
      */
     private void resetOffsets(final Set<TopicPartition> partitions) {
         final Map<TopicPartition, Long> offsetResets = new HashMap<>();
+        final Set<TopicPartition> toThrowFor = new HashSet<>();
```

Comment:
nit: maybe we could name this something more descriptive? Perhaps `partitionsWithNoOffsets`?